pandas/io/excel.py

```diff
@@ -1172,8 +1174,74 @@ def _convert_to_protection(cls, protection_dict):
         return Protection(**protection_dict)
 
 
-register_writer(_Openpyxl2Writer)
+register_writer(_Openpyxl20Writer)
 
+class _Openpyxl22Writer(_Openpyxl20Writer):
+    """
+    Note: Support for OpenPyxl v2.2 is currently EXPERIMENTAL (GH7565).
+    """
+    engine = 'openpyxl22'
+    openpyxl_majorver = 2
+
+    def write_cells(self, cells, sheet_name=None, startrow=0, startcol=0):
+        # Write the frame cells using openpyxl.
+        from openpyxl import styles
+
+        sheet_name = self._get_sheet_name(sheet_name)
+
+        if sheet_name in self.sheets:
+            wks = self.sheets[sheet_name]
+        else:
+            wks = self.book.create_sheet()
+            wks.title = sheet_name
+            self.sheets[sheet_name] = wks
+
+        for cell in cells:
+            xcell = wks.cell(row=startrow + cell.row + 1, column=startcol + cell.col + 1)
+            xcell.value = _conv_value(cell.val)
+
+            # Apply format codes before cell.style to allow override
+            if isinstance(cell.val, datetime.datetime):
+                xcell.number_format = self.datetime_format
+
+            elif isinstance(cell.val, datetime.date):
```

Comment:
this would almost never be true, as we only have internally ``Timestamp`` (which is a sub-class of ``datetime.datetime``). prob what you want is something like ``core.format._is_dates_only`` (which you should only call on an entire array/columns (a-priori to iterating over the cells)